packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -25,16 +30,26 @@ let portalObserver: MutationObserver | undefined;
 
 const portalObserverConfig: MutationObserverInit = {
     childList: true,
-    subtree: true,
 };
 
 function adoptChildNode(node: Node, fn: ShadowRootResolver, shadowToken: string | undefined) {
-    if (getShadowRootResolver(node) === fn) {
+    const previousNodeShadowResolver = getShadowRootResolver(node);
+    if (previousNodeShadowResolver === fn) {
         return; // nothing to do here, it is already correctly patched
     }
     setShadowRootResolver(node, fn);
     if (node instanceof Element) {
         setShadowToken(node, shadowToken);
+
+        if (isHostElement(node)) {
+            // Root LWC elements can't get content slotted into them, therefore we don't observe their children.
+            return;
+        }
+
+        if (isUndefined(previousNodeShadowResolver)) {
+            // we only care about Element without shadowResolver (no MO.observe has been called)
+            MutationObserverObserve.call(portalObserver, node, portalObserverConfig);
+        }
         // recursively patching all children as well
         const childNodes = getInternalChildNodes(node);
```

Comment:
@caridy, @ravijayaramappa since for portals we don't observe mutations on LWC components, what do you think about using the native apis? (`childNodesGetter`)

Reply:
Oh, that's a great idea. Since this element is not a host, the only other option is to be a slot, which we don't really care much since we are saying that slots are only good slots if they are coming from template.... we can just use `childNodesGetter` here to speed things up.